nelson-feng/wk04/2-tues/mta.rb

```diff
@@ -0,0 +1,87 @@
+# // # MTA
+n = ["Times Square", "34th", "28th", "23rd", "Union Square", "8th"]
+l = ["8th", "6th", "Union Square", "3rd", "1st"]
+six= ["Grand Central", "33rd", "28th", "23rd", "Union Square", "Astor Place"]
+def travel(ls,ss,le,se)
+    start = ls.index(ss)
+    finish = le.index(se)
+    stns = []
+    ls[start+1..finish -1].each { |i| stns << i } if (finish - start).positive? #If Positive then it will iterate left to right
+    ls[finish +1..start-1].reverse_each { |i| stns << i } if (finish - start).negative?
+    return stns
+end
+
+def before_union(ls,ss)
+    start = ls.index(ss)
+    start_union = ls.index("Union Square") # √√
+    stns = []
+    ls[start +1..start_union -1].each { |i| stns << i } if (start_union - start).positive?
+    ls[start_union+1..start-1].reverse_each { |i| stns << i } if (start_union - start).negative?
```

Comment:
I suspect the bug you mentioned comes from these lines, specifically the +`1` & `-1`'s.